scripts/benchmark-runner.py

```diff
@@ -191,9 +195,151 @@ def print_list(names):
+    ]
+    if not binaries:
+        raise ValueError(f"No binaries found at path '{binary_path.resolve()}'")
+
+    print(f"Found {len(binaries)} benchmark binaries")
+    return binaries
+
+
+def _default_binary_path():
+    repo_root = pathlib.Path(__file__).parent.parent.absolute()
+    return repo_root.joinpath("_build", "release", "velox", "benchmarks", "basic")
+
+
+def _normalize_path(binary_path: str) -> pathlib.Path:
+    path = pathlib.Path(binary_path)
+    if not path.is_absolute():
+        path = pathlib.Path.cwd().joinpath(path).resolve()
+    return path
+
+
+def run_all_benchmarks(
+    output_dir,
+    binary_path=None,
+    binary_filter=None,
+    bm_filter=None,
+    bm_max_secs=None,
+    bm_max_trials=None,
+    bm_estimate_time=False,
+):
+    if binary_path:
+        binary_path = _normalize_path(binary_path)
+    else:
+        binary_path = _default_binary_path()
+
+    binaries = _find_binaries(binary_path)
+    output_dir_path = pathlib.Path(output_dir)
+    output_dir_path.mkdir(parents=True, exist_ok=True)
+
+    for binary_path in binaries:
+        if binary_filter and not re.search(binary_filter, binary_path.name):
+            continue
+
+        out_path = output_dir_path / f"{binary_path.name}.json"
+        print(f"Executing and dumping results for '{binary_path}' to '{out_path}':")
+        run_command = [
+            binary_path,
+            "--bm_json_verbose",
+            out_path,
+        ]
+
+        if bm_max_secs:
+            run_command.extend(["--bm_max_secs", str(bm_max_secs)])
+
+        if bm_max_trials:
+            run_command.extend(["--bm_max_trials", str(bm_max_trials)])
+
+        if bm_filter:
+            run_command.extend(["--bm_regex", bm_filter])
+
+        if bm_estimate_time:
+            run_command.append("--bm_estimate_time")
+
+        try:
+            print(run_command)
+            subprocess.run(run_command, check=True)
+        except subprocess.CalledProcessError as e:
+            print(e.stderr.decode("utf-8"))
+            raise e
+
+
+def upload_results(output_dir, run_id):
+    print(f"Uploading results from {output_dir=} to Conbench with {run_id=}")
+
+    # Check there's actually results first
+    if not list(pathlib.Path(output_dir).rglob("*.json")):
+        print("No results to upload")
+        return
+
+    # Import benchadapt inside this function so you don't need it if you're just running benchmarks
+    from benchadapt.adapters import FollyAdapter
+    from benchadapt.log import log
+
+    # Print the POSTs in the logs
+    log.setLevel("DEBUG")
+
+    # benchadapt needs these for logging in to Conbench
+    required_env_vars = {"CONBENCH_URL", "CONBENCH_EMAIL", "CONBENCH_PASSWORD"}
+    missing_env_vars = required_env_vars - set(os.environ)
+    if missing_env_vars:
+        print(
+            "Not uploading results to Conbench because these env vars are missing: "
+            f"{missing_env_vars}"
```

Comment:
This error will happen on PRs until we figure out a more secure way to use secrets in PR builds.